org.opentosca.container.api/src/org/opentosca/container/api/controller/CsarController.java

```diff
@@ -320,6 +321,43 @@ public Response deleteCsar(@ApiParam("ID of CSAR") @PathParam("csar") final Stri
 
         return Response.noContent().build();
     }
+    
+    @POST
+    @Path("/transform")
+    @ApiOperation(value = "Transform this CSAR to a new CSAR")
+    @Consumes({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    public Response transformCsar(@ApiParam(required = true) final CsarTransformRequest request) {
+    	
+    	String sourceCsarName = request.getSourceCsarName();
+    	String targetCsarName = request.getTargetCsarName();
+    	
+    	
+    	
+    	CSARID csarId = this.csarService.generateTransformationPlans(new CSARID(sourceCsarName), new CSARID(targetCsarName));
+    	
+    	this.controlService.setDeploymentProcessStateStored(csarId);
+        boolean success = this.controlService.invokeTOSCAProcessing(csarId);
```

Comment:
can be inlined, successive writes are never read

Reply:
added proper response, redirect is not really possible as transformation plans are executed on service instances only and this method generate basically  a plan for all possible instances